docker/api/client.py

```diff
@@ -209,6 +218,20 @@ def __init__(self, base_url=None, version=None,
                 'library.'.format(MINIMUM_DOCKER_API_VERSION)
             )
 
+    def __exit__(self, exc_type, exc_value, traceback):
+        super().__exit__(exc_type, exc_value, traceback)
+        self._close_sockets()
```

Comment:
I'm somewhat unclear on the flow here. If we don't define `__exit__`, won't the parent's `__exit__` call our close method anyway, so the `close` definition later would be sufficient.

I might be overlooking something though.

Reply:
I also feel like in the current flow, we'll super-call close twice, once in `super().__exit__`, and then again in our own `close` called from here. That seems odd, and I see a double-close in one of the tests too. Is there some _other_ issue with `close` that we're covering up by recalling `close()` so much?